### PR TITLE
Add custom_randperm function which outputs the same random numbers than matlab's randperm function

### PR DESCRIPTION
**Pull request by jmontoyam**
Helper function to emulate the same behavior than matlab's randperm function.
This function is useful for comparing against matlab scripts which use the randperm function, for instance, for comparison against the eeglab infomax code.

### CODE REVIEW

File: mne/utils.py [hunk 1]
```diff
@@ -1777,3 +1777,45 @@ def _time_mask(times, tmin=None, tmax=None, strict=False):
...
+    uses the same matlab's random seed.
+    This function is useful for comparing against matlab scripts
+    which use the randperm function.
+    Note: randperm(n) matlab function generates a random sequence between
+    1 and n, whereas custom_randperm(n, rng) function generates a random
+    sequence between 0 and n-1, that is:
+    randperm(n) = custom_randperm(n, rng) - 1
+
+    Parameters
+        ----------
```
Comment (dengemann) [inline mne/utils.py]: something looks wrong here, please correct the alignment

File: mne/utils.py [hunk 2]
```diff
@@ -1777,3 +1777,46 @@ def _time_mask(times, tmin=None, tmax=None, strict=False):
         mask |= isclose(times, tmin)
         mask |= isclose(times, tmax)
     return mask
+
+
+def custom_randperm(n, rng):
```
Comment (agramfort) [inline mne/utils.py]: random_permutation(n_samples, random_state=None)

should be the signature.

File: mne/utils.py [hunk 3]
```diff
@@ -1777,3 +1777,46 @@ def _time_mask(times, tmin=None, tmax=None, strict=False):
...
+        Randomly permuted sequence between 0 and n-1.
+
+    Example
+    -------
+    >>> rng = np.random.RandomState(42)
+    >>> p = custom_randperm(10, rng)  # permuted sequence between 0 and 9
+    >>> p
+    array([7, 6, 5, 1, 4, 9, 10, 3, 8, 2])
+    """
+    idx = rng.rand(n)
```
Comment (agramfort) [inline mne/utils.py]: before do:

rng = check_random_state(random_state)

**Comment by agramfort** [comment]
travis is not happy. See why.

Besides LGTM

**Comment by Eric89GXL** [comment]
Look into the `# doctest:` comments we have in our code, they cause the doctests to be more lenient with tests (or ignore outputs altogether). You can probably do `# doctest: +SKIP` on this one.

**Comment by agramfort** [comment]
thanks @jmontoyam !

**Pull request merged**